bin/dry-run.rb

```diff
@@ -447,15 +454,19 @@ def handle_dependabot_error(error:, dependency:)
 always_clone = Dependabot::Utils.
                always_clone_for_package_manager?($package_manager)
 if $options[:clone] || always_clone
-  $repo_contents_path = Dir.mktmpdir
-  puts "=> cloning into #{$repo_contents_path}"
+  $repo_contents_path = File.expand_path(File.join("tmp", $repo_name.split("/")))
```

Comment:
Concern: since the `FileFetcher` [exits](https://github.com/dependabot/dependabot-core/blob/e7c1929d7f850f8516311cc41753958cb6b980d2/common/lib/dependabot/file_fetchers/base.rb#L476) if the path exists, this flow won't:
* fetch the latest from the repository
* update if `--branch` or `--commit` flags are provided

Is that expected?